raiden/tests/scenarios/join-network-scenario-J4.yaml

```diff
@@ -0,0 +1,46 @@
+settings:
+  gas_price: "fast"
+
+
+token:
+
+nodes:
+  commands:
+    stop: "pkill -f '127.0.0.1:50{:02d}'"
+    kill: "pkill -9 -f '127.0.0.1:50{:02d}'"
+    start: "raiden --config-file raiden_{0:02d}.toml --api-address 127.0.0.1:50{0:02d}"
+  range:
+    first: 1
+    last: 4
+    template: "localhost:50{:02d}"
+
+scenario:
+  serial:
+    tasks:
+      - parallel:
+          name: "Setting up a network"
+          tasks:
+            - open_channel: {from: 0, to: 1, total_deposit: 10, expected_http_status: 201}
+            - open_channel: {from: 0, to: 2, total_deposit: 10, expected_http_status: 201}
+            - open_channel: {from: 1, to: 2, total_deposit: 10, expected_http_status: 201}
+
+      - serial:
+          name: "Checking the network"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 10, balance: 10, state: "opened"}
+            - assert: {from: 0, to: 2, total_deposit: 10, balance: 10, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 10, balance: 10, state: "opened"}
+
+      - serial:
+          name: "Node Nr. 4 joins a network after having left it and after settlement of all channels took place"
+          tasks:
+            - join_network: {from: 3, funds: 100, initial_channel_target: 3, joinable_funds_target: 0.4, expected_http_status: 204}
+            - wait: 10
+            - leave_network: {from: 3}
+            - assert: {from: 3, to: 0, expected_http_status: 404}
```

Comment:
why is this?

Reply:
because `leave_network` closes and settles all channels, you will get a 404 response trying to reach those after you have left the network. Makes sense?

Reply:
Ah sure, it blocks until all is settled. Sorry for the noise.